### DISCUSSION TITLE
Failed to resolve tileset descriptors: Loading request canceled

### DISCUSSION THREAD

**Discussion by devWaleed** [rnmapbox/maps#2577] [Q&A]
I have the following asynchronous code for download map regions. This function is called on multiple regions/farm-fields as well as user location asynchronously, such that all regions are downloaded in parallel. In debug/development mode I rarely get this error. But in production release app I get this issue sometimes and the whole offline user experience is ruined. Notice that for users and fields same function is being used to download regions but it fails mostly for fields and not for users.

**Download Pack helper function:**

```
const downloadPack = async (packName, bounds) => {
  // Delete old pack
  await MapboxGL.offlineManager.deletePack(packName);

  const progressListener = (offlineRegion, status) => {
    if (status.state === "complete") {
      console.log(
        `Pack: ${packName}`,
        formatBytes(status.completedResourceSize),
        status.percentage,
        status.state,
      );
    }
  };

  const errorListener = (offlineRegion, err) => {
    console.log("pack error", offlineRegion, err);
  };

  const packConfig = {
    name: packName,
    styleURL: styleURL,
    minZoom: MAP_CONFIG.MIN_ZOOM_ALLOWED,
    maxZoom: MAP_CONFIG.MAX_ZOOM_ALLOWED,
    bounds,
  };

  await MapboxGL.offlineManager.createPack(
    packConfig,
    progressListener,
    errorListener,
  );

  // console.log("packConfig", packConfig);
};
```

**Errors:**
```
  LOG  pack error {"_metadata": null, "pack": {"bounds": "{\"type\":\"FeatureCollection\",\"features\":[{\"type\":\"Feature\",\"properties\":{},\"geometry\":{\"type\":\"Point\",\"coordinates\":[67.05473691050432,24.859774187835313]}},{\"type\":\"Feature\",\"properties\":{},\"geometry\":{\"type\":\"Point\",\"coordinates\":[67.05879400202377,24.864996599470516]}}]}", "metadata": "{
  \"name\" : \"field-region-74\"
}"}} {"message": "Failed to resolve tileset descriptors: Loading request canceled", "name": "field-region-74"}
 LOG  pack error {"_metadata": null, "pack": {"bounds": "{\"type\":\"FeatureCollection\",\"features\":[{\"type\":\"Feature\",\"properties\":{},\"geometry\":{\"type\":\"Point\",\"coordinates\":[67.05603555964123,24.863974062407152]}},{\"type\":\"Feature\",\"properties\":{},\"geometry\":{\"type\":\"Point\",\"coordinates\":[67.06011966003771,24.86902696726922]}}]}", "metadata": "{
  \"name\" : \"field-region-73\"
}"}} {"message": "Failed to resolve tileset descriptors: Loading request canceled", "name": "field-region-73"}
 LOG  pack error {"_metadata": null, "pack": {"bounds": "{\"type\":\"FeatureCollection\",\"features\":[{\"type\":\"Feature\",\"properties\":{},\"geometry\":{\"type\":\"Point\",\"coordinates\":[67.05860259878298,24.868365914318616]}},{\"type\":\"Feature\",\"properties\":{},\"geometry\":{\"type\":\"Point\",\"coordinates\":[67.06177770104551,24.871585472853567]}}]}", "metadata": "{
  \"name\" : \"field-region-72\"
}"}} {"message": "Failed to resolve tileset descriptors: Loading request canceled", "name": "field-region-72"}
 LOG  pack error {"_metadata": null, "pack": {"bounds": "{\"type\":\"FeatureCollection\",\"features\":[{\"type\":\"Feature\",\"properties\":{},\"geometry\":{\"type\":\"Point\",\"coordinates\":[67.0527601437771,24.865385527088804]}},{\"type\":\"Feature\",\"properties\":{},\"geometry\":{\"type\":\"Point\",\"coordinates\":[67.06248793623786,24.878936670953124]}}]}", "metadata": "{
  \"name\" : \"field-region-71\"
}"}} {"message": "Failed to resolve tileset descriptors: Loading request canceled", "name": "field-region-71"}

 LOG  Pack: field-region-70 7.52 MB 100 complete
 LOG  Pack: user-region-882 1.23 MB 100 complete
 LOG  Pack: user-region-881 1.23 MB 100 complete
 LOG  Pack: user-region-862 1.23 MB 100 complete
 LOG  Pack: user-region-883 5.25 MB 100 complete
```

**Comment by devWaleed** [comment]
Notice that one of the fields field-region-70 area was downloaded successfully.

**Reply by mfazekas** [reply]
Pls try to make a reproducible example - see https://github.com/rnmapbox/maps/issues/2537 then there is a chance that we can debug/fix the issue.

Also make sure you're testing with a recent release as https://github.com/rnmapbox/maps/pull/2533  that is beta64 or more recent - https://github.com/rnmapbox/maps/releases/tag/v10.0.0-beta.64

**Reply by devWaleed** [reply]
I have provided the helper function which I have and the error it gives me. From the error NE/SW coordinates are also available. I understand that it doesn't produce this error everytime. Sometimes after restarting the app or reinstalling the app fixes the issue. Can you please tell me how I can get the error that you have in #2537 ? The above error is all I have. 

I am using beta-65 as of now.

If you can tell me how I can get more detailed errors, perhaps from the native side. 

Does the app limit to 750 tiles per user? or 750 packs ? How can I get this limit increased for my use case.

**Reply by devWaleed** [reply]
@mfazekas do you think the issue is with async pack creation or the tiles limit? 

I have a very small set of tiles which I am downloading for offline in my development environment right now. Between Zoom 11 and 14 and it is giving me this error. I really don't think I have exceeded 750 tiles limit. How do I check this?

can you point me into some direction please?

**Comment by devWaleed** [comment]
I had a discussion with Mapbox support team and according to them map tiles limit can be increased as per our needs. I checked this package documentation and found: setTileCountLimit. I tried this function and it gives me error 'Not implemented'. 

I have also changed my code to download all offline regions in synchronous manner one-by-one and not in parallel. I really doubt it crosses 750 tiles limit but still after this synchronous change, i am facing this error. I am unable to crease the tile limit count as well due to function not implemented yet.

@mfazekas
